File: php/testcode/weakcipher_029.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_weakcipher_md5_derived_key
function weakcipher029(BenchmarkRequest $req): BenchmarkResponse {
    $password = $req->param('pass');
    $key = md5($password); // vuln-code-snippet vuln-line php_weakcipher_md5_derived_key
    $enc = openssl_encrypt($req->param('data'), 'AES-256-CBC', $key, 0, random_bytes(16));
    return BenchmarkResponse::ok($enc);
}
// vuln-code-snippet end php_weakcipher_md5_derived_key

File: ruby/testcode/weakcipher_029.rb
require_relative 'shared'
require 'openssl'

# vuln-code-snippet start ruby_weakcipher_aes128_ecb
def encrypt_aes128_ecb(req)
  plaintext = req.body_str
  cipher = OpenSSL::Cipher.new('AES-128-ECB') # vuln-code-snippet vuln-line ruby_weakcipher_aes128_ecb
  cipher.encrypt
  cipher.key = cipher.random_key
  ciphertext = cipher.update(plaintext) + cipher.final
  BenchmarkResponse.ok(ciphertext.unpack1('H*'))
end
# vuln-code-snippet end ruby_weakcipher_aes128_ecb
